wandb/proto/wandb_internal.proto

```diff
@@ -68,6 +68,7 @@ message Control {
   bool flow_control = 6;     // message should be passed to flow control
   int64 end_offset = 7;      // end of message offset of this written message
   string connection_id = 8;  // connection id
+  bool internal = 9;         // message was created by an internal process
```

Comment:
Is this used for client<->service communication? We should really avoid using protos for anything other than that.

Even if it's unavoidable, the description is too vague. What constitutes an internal process? What happens to "internal" metric records?

Reply:
> in this case it is not really possible to follow the client<->service communication as the record actually created in the internal process

Right, that's what I'm saying we should not use the proto for.

We do have a higher level way of passing around information: the Work interface defined in core/internal/runwork/work.go. Right now, this is primarily a wrapper for Records via WorkRecord which delegates to the monolithic Handler and Sender functions, but the goal of it is to allow factoring out implementations for individual records and to pass around internal state.

I can follow up your PR with an example of how to do it. For now, can you at least move the `internal` field to the MetricRecord proto instead? It doesn't apply outside of it.

Reply:
> For now, can you at least move the internal field to the MetricRecord proto instead? It doesn't apply outside of it.

I think i tried it and it didn't work correctly, can try again, but theoretically it can be useful for system metrics and other records as well?

Reply:
It would be reasonable to add an `expanded_from_glob` field to `MetricRecord`. Can you say what didn't work?

> theoretically it can be useful for system metrics and other records as well?

System metrics are always generated, so `internal` would always be `true`. We shouldn't treat records in the transaction log differently based on where they came from; expanded metrics are an exception to work around server limitations (in an ideal world, we wouldn't store expanded metrics in the transaction log at all).

Reply:
yeah sounds good i can add it to the metric record

Reply:
> (in an ideal world, we wouldn't store expanded metrics in the transaction log at all)
agreed!